clients/src/main/java/org/apache/kafka/common/security/oauthbearer/internals/secured/HttpAccessTokenRetriever.java

```diff
@@ -240,6 +240,9 @@ static String handleOutput(final HttpURLConnection con) throws IOException {
         int responseCode = con.getResponseCode();
         log.debug("handleOutput - responseCode: {}", responseCode);
 
+        // NOTE: the contents of the response should not be logged so that we don't leak any
+        // sensitive data.
+        // TODO: is it OK to log the error response body and/or its formatted version?
```

Comment:
@smjn @omkreddy: open question. Is it safe to log the `errorResponseBody` in whole or in part?

The `formatErrorMessage` method attempts to extract the core error message out of the `errorResponseBody`, but it falls back to returning the whole string verbatim if it can't parse or find the error message in the response.

Thoughts?

Reply:
Hi I had referred the rfc which mentioned about the error response - https://www.ietf.org/rfc/rfc6749.txt , section 5.2

P.S. has there been any instance of info leak? There are only 3 things to leak - client_id, secret and scope. Of this only the secret is problematic. As per the rfc the value of the attribute is not printed.

Reply:
@smjn - No known instance that I'm aware of. A security review brought it to my attention, so addressing preemptively.